### PR TITLE
PUPIL-277 organise components from download and share components folder

### PR DESCRIPTION
**Pull request by weronika-szalas**
## Description

Music year: 1982

Move/rename/extract the following components from `DownloandsAndShareComponents` and `Downloads` folders:
-`ResourcePageLayout`
- `CopyrightNotice`
- `NoResourcesToDownload`
- `NoResourcesToShare`
- `DownloadCardGroup`
- `ShareCardGroup`
- `ResourceCardGroup`
- `DownloadConfirmation`
- `DetailsCompleted` 
- `DownloadConfirmation`
- `NextLessonCard`
- `SchoolDetails`
- `ShareLinks`
- `TermsAndConditionsCheckbox`
- move and rename hooks ann helpers folders from DownloadAnShareComponents
- move `ExpandingContainer` to `ArchivedComponents`

## Issue(s)

Fixes #PUPIL-277

### CODE REVIEW

File: src/components/ExpandingContainer/ExpandingContainer.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React, { FC, useState } from "react";
 
-import { containerTitleToPreselectMap } from "@/components/DownloadAndShareComponents/helpers/containerTitleToPreselectMap";
+import { containerTitleToPreselectMap } from "@/components/TeacherComponents/helpers/downloadAndShareHelpers/containerTitleToPreselectMap";
```
Comment (njblore) [inline src/components/ExpandingContainer/ExpandingContainer.tsx:3]: feels a bit unpleasant for a generic component to import from a squad-specific component, should one or the other be moved?

Comment (weronika-szalas) [reply src/components/ExpandingContainer/ExpandingContainer.tsx:3]: yes! thanks for spotting - I will update now

Comment (weronika-szalas) [reply src/components/ExpandingContainer/ExpandingContainer.tsx:3]: I moved `ExpandingContainer` to `ArchivedComponents` aa it is not used anywhere

**Review by njblore — APPROVED** [review]

**Pull request merged**